pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -375,18 +419,20 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 	switch tls.Mode {
 	case networking.TLSSettings_DISABLE:
 		// TODO: Need to make sure that authN does not override this setting
+		// We remove the TlsContext because it can be written because of configmap.MTLS settings.
+		cluster.TlsContext = nil
 	case networking.TLSSettings_SIMPLE:
 		cluster.TlsContext = &auth.UpstreamTlsContext{
 			CommonTlsContext: &auth.CommonTlsContext{
 				ValidationContext: certValidationContext,
 			},
 			Sni: tls.Sni,
 		}
+	case networking.TLSSettings_MUTUAL, networking.TLSSettings_ISTIO_MUTUAL:
 		if cluster.Http2ProtocolOptions != nil {
```

Comment:
Delete this block. It should be set after cluster.TlsContext (and already exist below)